pkg/credentialprovider/keyring.go

```diff
@@ -96,6 +97,20 @@ type SecretCoordinates struct {
 	Name      string
 }
 
+// ServiceAccountTokenSource is used to provide a service account token
+type ServiceAccountTokenSource struct {
+	Audience string
+
+	Namespace                 string
+	ServiceAccountName        string
+	ServiceAccountUID         string
+	ServiceAccountAnnotations map[string]string
+	PodName                   string
+	PodUID                    string
+	ServiceAccountTokenHash   string // SHA256 hash of the service account token
+	CacheType                 string
```

Comment:
suggest dropping these and saying we enforce shared access to pulled images based on the service account dimensions (plus audience and annotations that inform the credential provider)